### DISCUSSION TITLE
Write over existing pdf template

### DISCUSSION THREAD

**Discussion by Dartths** [QuestPDF/QuestPDF#130] [Q&A] [answered]
Is there any way to read in an existing pdf and work with it, use case is trying to fill in a pdf form template that I have. From what I can see the closest thing to working with pre-existing files is to use images and layers to write over them. 
Any ideas or suggestion would be great, thanks!

**Comment by MarcinZiabek** [comment, marked as answer]
Hello 😁 I am afraid that at this moment it is impossible to add content on top of the already existing document. The current library scope is PDF generation only. Have a great day!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
